### ISSUE TITLE
SINGLE_METAL_MOFS_ONLY and no-metal MOFs

### ISSUE THREAD

**Issue by Andrew-S-Rosen** [tobacco-mofs/tobacco_3.0#7]
When `SINGLE_METAL_MOFS_ONLY = False`, I believe you can get a "MOF" with no metals if it pulls an organic node and edge. I unfortunately am having trouble finding which combination got me this so this can be closed if not reproducible, but there should be a check to ensure at least 1 metal is present if there's not that already.

**Comment by rytheranderson** [comment]
I checked some examples, I do not think it is possible, the current check is to convert the list of all metals in the MOF to a set and check that the length is exactly one, so I will close for now, but let me know if you can reproduce the problem.

**Comment by Andrew-S-Rosen** [comment]
Hi @rytheranderson. As an FYI: if the edge is `ntn_edge.cif` and you have an inorganic and organic node, it is possible to get a "MOF" where the vertices are only the organic node connected to one another when `SINGLE_METAL_MOFS_ONLY = False`. I can provide an example if desired. I don't know that this is _super_ necessary to resolve, as it's technically doing what the user is asking for -- trying all combinations. It just so happens that one of those combinations is all-organic.

**Comment by rytheranderson** [comment]
Hi Andrew, 

I revisited this today since others seem to be having issues with the SINGLE_METAL_MOFS_ONLY flag, I think the problem was the documentation was not clear, I intended that SINGLE_METAL_MOFS_ONLY would filter out both no-metal structures and multi-metal structures, so if it was set to false, any structure could be written (including COFs).

This worked fine for me since I did not want multi-metal MOFs or COFs, but there are instances where one wants multi-metal MOFs without COFs, so I added a new flag MOFS_ONLY which just filters out any structures without metals. So if SINGLE_METAL_MOFS_ONLY is True only structures with exactly one metal are written, if MOFS_ONLY is true only structures with >0 metals are written, and if both are false any structure can be written.
